DiffMatchPatch.zig

```diff
@@ -2,9 +2,32 @@ const DiffMatchPatch = @This();
 
 const std = @import("std");
 const testing = std.testing;
+const Allocator = std.mem.Allocator;
 const ArrayListUnmanaged = std.ArrayListUnmanaged;
 const DiffList = ArrayListUnmanaged(Diff);
 
+/// Deinit an `ArrayListUnmanaged(Diff)` and the allocated slices of
+/// text in each `Diff`.
+pub fn deinitDiffList(allocator: Allocator, diffs: *DiffList) void {
+    defer diffs.deinit(allocator);
+    for (diffs.items) |d| {
+        allocator.free(d.text);
+    }
+}
+
+fn freeRangeDiffList(
+    allocator: Allocator,
+    diffs: *DiffList,
+    start: usize,
+    len: usize,
+) void {
+    const after_range = start + len;
+    const range = diffs.items[start..after_range];
```

Comment:
nit: Zig has a special syntax to slice-by-length.
```suggestion
    const range = diffs.items[start..][0..len];
```